### PR TITLE
SDIT-1881 Add CSIP Updated event handling and mapping failure tests for create

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPFactorSynchronisationIntTest.kt [hunk 1]
```diff
@@ -297,12 +302,304 @@ class CSIPFactorSynchronisationIntTest : SqsIntegrationTestBase() {
           }
         }
       }
+
+      @Nested
+      @DisplayName("When mapping POST fails")
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPFactorSynchronisationIntTest.kt:307]: Mapping tests from previous PR (sorry, should've just done as a separate PR)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPFactorSynchronisationIntTest.kt [hunk 2]
```diff
@@ -297,12 +302,304 @@ class CSIPFactorSynchronisationIntTest : SqsIntegrationTestBase() {
...
+              )
+            }
+          }
+        }
+      }
+    }
+  }
+
+  @Nested
+  @DisplayName("CSIP_FACTORS-UPDATED")
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPFactorSynchronisationIntTest.kt:447]: New tests for the updated event

**Review by petergphillips — APPROVED** [review]
LGTM
